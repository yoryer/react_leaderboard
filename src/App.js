import React, { Component } from 'react';
import orderBy from 'lodash/orderBy';

import Age from './components/Age';
import Name from './components/Name';
import Points from './components/Points';
import Rank from './components/Rank';
import Table from './components/Table';
import usersJSON from './data.json';

export default class App extends Component {
  state = {
    users: usersJSON,
    columnToSort: '',
    sortDirection: 1,
  };

  handleSort = (columnName) => {
    this.setState(state => ({
      columnToSort: columnName,
      sortDirection: state.columnToSort === columnName ? state.sortDirection * -1 : 1,
    }));
  }

  render() {
    return (
      <div className="text-center buttons">
        <header className="text-center">
          <h1>Leaderboard</h1>
        </header>
        <div className="text-center buttons">
          <Age handleSort={this.handleSort} columnToSort={this.state.columnToSort} sortDirection={this.state.sortDirection} />
          <Name handleSort={this.handleSort} columnToSort={this.state.columnToSort} sortDirection={this.state.sortDirection} />
          <Points handleSort={this.handleSort} columnToSort={this.state.columnToSort} sortDirection={this.state.sortDirection} />
          <Rank handleSort={this.handleSort} columnToSort={this.state.columnToSort} sortDirection={this.state.sortDirection} />
          <Table data={orderBy(this.state.users, this.state.columnToSort, this.state.sortDirection === 1 ? 'asc' : 'desc',)} />
        </div>
      </div>
    );
  }
}
