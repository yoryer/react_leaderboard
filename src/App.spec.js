import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import App from './App';

afterEach(() => {
  cleanup();
});

test('Age is sorted correctly', () => {
  const { getByText, getByTestId } = render(<App />);

  const age = getByText('Age', { selector: 'button' });

  fireEvent.click(age);

  const table = getByTestId('leaderboard-table');
  expect(table).toMatchSnapshot();
});

test('Name is sorted correctly', () => {
  const { getByText, getByTestId } = render(<App />);

  const name = getByText('Name', { selector: 'button' });
  fireEvent.click(name);

  const table = getByTestId('leaderboard-table');
  expect(table).toMatchSnapshot();
});

test('Points is sorted correctly', () => {
  const { getByText, getByTestId } = render(<App />);

  const points = getByText('Points', { selector: 'button' });

  fireEvent.click(points);

  const table = getByTestId('leaderboard-table');
  expect(table).toMatchSnapshot();
});

test('Rank is sorted correctly', () => {
  const { getByText, getByTestId } = render(<App />);

  const age = getByText('Rank', { selector: 'button' });
  fireEvent.click(age);

  const table = getByTestId('leaderboard-table');
  expect(table).toMatchSnapshot();
});
