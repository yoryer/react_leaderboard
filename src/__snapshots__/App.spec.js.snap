// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Age is sorted correctly 1`] = `
<table
  class="table table-striped"
  data-testid="leaderboard-table"
>
  <thead>
    <tr>
      <th>
        Age
      </th>
      <th>
        Name
      </th>
      <th>
        Points
      </th>
      <th>
        Rank
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        10
      </td>
      <td>
        Michael E. Rogers
      </td>
      <td>
        274
      </td>
      <td>
        7
      </td>
    </tr>
    <tr>
      <td>
        11
      </td>
      <td>
        Richard M. Mass
      </td>
      <td>
        312
      </td>
      <td>
        14
      </td>
    </tr>
    <tr>
      <td>
        13
      </td>
      <td>
        Joan T. Covert
      </td>
      <td>
        3124
      </td>
      <td>
        13
      </td>
    </tr>
    <tr>
      <td>
        15
      </td>
      <td>
        Elizabeth R. Olsen
      </td>
      <td>
        452
      </td>
      <td>
        9
      </td>
    </tr>
    <tr>
      <td>
        17
      </td>
      <td>
        Camille B. Burton
      </td>
      <td>
        654645
      </td>
      <td>
        15
      </td>
    </tr>
    <tr>
      <td>
        18
      </td>
      <td>
        Tommye R. Robinson
      </td>
      <td>
        5345345
      </td>
      <td>
        4
      </td>
    </tr>
    <tr>
      <td>
        19
      </td>
      <td>
        Grace C. Smith
      </td>
      <td>
        342
      </td>
      <td>
        17
      </td>
    </tr>
    <tr>
      <td>
        20
      </td>
      <td>
        Jennifer C. West
      </td>
      <td>
        213
      </td>
      <td>
        22
      </td>
    </tr>
    <tr>
      <td>
        21
      </td>
      <td>
        John L. Sandlin
      </td>
      <td>
        34242
      </td>
      <td>
        25
      </td>
    </tr>
    <tr>
      <td>
        22
      </td>
      <td>
        Joseph J. Freeman
      </td>
      <td>
        354
      </td>
      <td>
        8
      </td>
    </tr>
    <tr>
      <td>
        23
      </td>
      <td>
        Joseph S. Perez
      </td>
      <td>
        31
      </td>
      <td>
        20
      </td>
    </tr>
    <tr>
      <td>
        24
      </td>
      <td>
        John R. Green
      </td>
      <td>
        6546
      </td>
      <td>
        19
      </td>
    </tr>
    <tr>
      <td>
        25
      </td>
      <td>
        Sharon R. Ryan
      </td>
      <td>
        134
      </td>
      <td>
        1
      </td>
    </tr>
    <tr>
      <td>
        29
      </td>
      <td>
        Arlene R. Grimes
      </td>
      <td>
        7675
      </td>
      <td>
        6
      </td>
    </tr>
    <tr>
      <td>
        30
      </td>
      <td>
        Stacey D. Ward
      </td>
      <td>
        64563
      </td>
      <td>
        24
      </td>
    </tr>
    <tr>
      <td>
        32
      </td>
      <td>
        Thomas M. Hawkins
      </td>
      <td>
        42342
      </td>
      <td>
        5
      </td>
    </tr>
    <tr>
      <td>
        33
      </td>
      <td>
        Dorothy J. Carroll
      </td>
      <td>
        42342
      </td>
      <td>
        18
      </td>
    </tr>
    <tr>
      <td>
        35
      </td>
      <td>
        George K. Pancoast
      </td>
      <td>
        755463
      </td>
      <td>
        2
      </td>
    </tr>
    <tr>
      <td>
        37
      </td>
      <td>
        Alice J. Goldsmith
      </td>
      <td>
        105
      </td>
      <td>
        12
      </td>
    </tr>
    <tr>
      <td>
        39
      </td>
      <td>
        Felix M. Seeber
      </td>
      <td>
        103596
      </td>
      <td>
        11
      </td>
    </tr>
    <tr>
      <td>
        40
      </td>
      <td>
        Robert O. Lambert
      </td>
      <td>
        18978
      </td>
      <td>
        16
      </td>
    </tr>
    <tr>
      <td>
        41
      </td>
      <td>
        Marlana J. Huff
      </td>
      <td>
        3123
      </td>
      <td>
        3
      </td>
    </tr>
    <tr>
      <td>
        42
      </td>
      <td>
        Susan D. Jimenez
      </td>
      <td>
        45345
      </td>
      <td>
        10
      </td>
    </tr>
    <tr>
      <td>
        43
      </td>
      <td>
        Margarita D. Lambert
      </td>
      <td>
        52342
      </td>
      <td>
        21
      </td>
    </tr>
    <tr>
      <td>
        65
      </td>
      <td>
        Louis E. McCrae
      </td>
      <td>
        1241
      </td>
      <td>
        23
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Name is sorted correctly 1`] = `
<table
  class="table table-striped"
  data-testid="leaderboard-table"
>
  <thead>
    <tr>
      <th>
        Age
      </th>
      <th>
        Name
      </th>
      <th>
        Points
      </th>
      <th>
        Rank
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        37
      </td>
      <td>
        Alice J. Goldsmith
      </td>
      <td>
        105
      </td>
      <td>
        12
      </td>
    </tr>
    <tr>
      <td>
        29
      </td>
      <td>
        Arlene R. Grimes
      </td>
      <td>
        7675
      </td>
      <td>
        6
      </td>
    </tr>
    <tr>
      <td>
        17
      </td>
      <td>
        Camille B. Burton
      </td>
      <td>
        654645
      </td>
      <td>
        15
      </td>
    </tr>
    <tr>
      <td>
        33
      </td>
      <td>
        Dorothy J. Carroll
      </td>
      <td>
        42342
      </td>
      <td>
        18
      </td>
    </tr>
    <tr>
      <td>
        15
      </td>
      <td>
        Elizabeth R. Olsen
      </td>
      <td>
        452
      </td>
      <td>
        9
      </td>
    </tr>
    <tr>
      <td>
        39
      </td>
      <td>
        Felix M. Seeber
      </td>
      <td>
        103596
      </td>
      <td>
        11
      </td>
    </tr>
    <tr>
      <td>
        35
      </td>
      <td>
        George K. Pancoast
      </td>
      <td>
        755463
      </td>
      <td>
        2
      </td>
    </tr>
    <tr>
      <td>
        19
      </td>
      <td>
        Grace C. Smith
      </td>
      <td>
        342
      </td>
      <td>
        17
      </td>
    </tr>
    <tr>
      <td>
        20
      </td>
      <td>
        Jennifer C. West
      </td>
      <td>
        213
      </td>
      <td>
        22
      </td>
    </tr>
    <tr>
      <td>
        13
      </td>
      <td>
        Joan T. Covert
      </td>
      <td>
        3124
      </td>
      <td>
        13
      </td>
    </tr>
    <tr>
      <td>
        21
      </td>
      <td>
        John L. Sandlin
      </td>
      <td>
        34242
      </td>
      <td>
        25
      </td>
    </tr>
    <tr>
      <td>
        24
      </td>
      <td>
        John R. Green
      </td>
      <td>
        6546
      </td>
      <td>
        19
      </td>
    </tr>
    <tr>
      <td>
        22
      </td>
      <td>
        Joseph J. Freeman
      </td>
      <td>
        354
      </td>
      <td>
        8
      </td>
    </tr>
    <tr>
      <td>
        23
      </td>
      <td>
        Joseph S. Perez
      </td>
      <td>
        31
      </td>
      <td>
        20
      </td>
    </tr>
    <tr>
      <td>
        65
      </td>
      <td>
        Louis E. McCrae
      </td>
      <td>
        1241
      </td>
      <td>
        23
      </td>
    </tr>
    <tr>
      <td>
        43
      </td>
      <td>
        Margarita D. Lambert
      </td>
      <td>
        52342
      </td>
      <td>
        21
      </td>
    </tr>
    <tr>
      <td>
        41
      </td>
      <td>
        Marlana J. Huff
      </td>
      <td>
        3123
      </td>
      <td>
        3
      </td>
    </tr>
    <tr>
      <td>
        10
      </td>
      <td>
        Michael E. Rogers
      </td>
      <td>
        274
      </td>
      <td>
        7
      </td>
    </tr>
    <tr>
      <td>
        11
      </td>
      <td>
        Richard M. Mass
      </td>
      <td>
        312
      </td>
      <td>
        14
      </td>
    </tr>
    <tr>
      <td>
        40
      </td>
      <td>
        Robert O. Lambert
      </td>
      <td>
        18978
      </td>
      <td>
        16
      </td>
    </tr>
    <tr>
      <td>
        25
      </td>
      <td>
        Sharon R. Ryan
      </td>
      <td>
        134
      </td>
      <td>
        1
      </td>
    </tr>
    <tr>
      <td>
        30
      </td>
      <td>
        Stacey D. Ward
      </td>
      <td>
        64563
      </td>
      <td>
        24
      </td>
    </tr>
    <tr>
      <td>
        42
      </td>
      <td>
        Susan D. Jimenez
      </td>
      <td>
        45345
      </td>
      <td>
        10
      </td>
    </tr>
    <tr>
      <td>
        32
      </td>
      <td>
        Thomas M. Hawkins
      </td>
      <td>
        42342
      </td>
      <td>
        5
      </td>
    </tr>
    <tr>
      <td>
        18
      </td>
      <td>
        Tommye R. Robinson
      </td>
      <td>
        5345345
      </td>
      <td>
        4
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Points is sorted correctly 1`] = `
<table
  class="table table-striped"
  data-testid="leaderboard-table"
>
  <thead>
    <tr>
      <th>
        Age
      </th>
      <th>
        Name
      </th>
      <th>
        Points
      </th>
      <th>
        Rank
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        23
      </td>
      <td>
        Joseph S. Perez
      </td>
      <td>
        31
      </td>
      <td>
        20
      </td>
    </tr>
    <tr>
      <td>
        37
      </td>
      <td>
        Alice J. Goldsmith
      </td>
      <td>
        105
      </td>
      <td>
        12
      </td>
    </tr>
    <tr>
      <td>
        25
      </td>
      <td>
        Sharon R. Ryan
      </td>
      <td>
        134
      </td>
      <td>
        1
      </td>
    </tr>
    <tr>
      <td>
        20
      </td>
      <td>
        Jennifer C. West
      </td>
      <td>
        213
      </td>
      <td>
        22
      </td>
    </tr>
    <tr>
      <td>
        10
      </td>
      <td>
        Michael E. Rogers
      </td>
      <td>
        274
      </td>
      <td>
        7
      </td>
    </tr>
    <tr>
      <td>
        11
      </td>
      <td>
        Richard M. Mass
      </td>
      <td>
        312
      </td>
      <td>
        14
      </td>
    </tr>
    <tr>
      <td>
        19
      </td>
      <td>
        Grace C. Smith
      </td>
      <td>
        342
      </td>
      <td>
        17
      </td>
    </tr>
    <tr>
      <td>
        22
      </td>
      <td>
        Joseph J. Freeman
      </td>
      <td>
        354
      </td>
      <td>
        8
      </td>
    </tr>
    <tr>
      <td>
        15
      </td>
      <td>
        Elizabeth R. Olsen
      </td>
      <td>
        452
      </td>
      <td>
        9
      </td>
    </tr>
    <tr>
      <td>
        65
      </td>
      <td>
        Louis E. McCrae
      </td>
      <td>
        1241
      </td>
      <td>
        23
      </td>
    </tr>
    <tr>
      <td>
        41
      </td>
      <td>
        Marlana J. Huff
      </td>
      <td>
        3123
      </td>
      <td>
        3
      </td>
    </tr>
    <tr>
      <td>
        13
      </td>
      <td>
        Joan T. Covert
      </td>
      <td>
        3124
      </td>
      <td>
        13
      </td>
    </tr>
    <tr>
      <td>
        24
      </td>
      <td>
        John R. Green
      </td>
      <td>
        6546
      </td>
      <td>
        19
      </td>
    </tr>
    <tr>
      <td>
        29
      </td>
      <td>
        Arlene R. Grimes
      </td>
      <td>
        7675
      </td>
      <td>
        6
      </td>
    </tr>
    <tr>
      <td>
        40
      </td>
      <td>
        Robert O. Lambert
      </td>
      <td>
        18978
      </td>
      <td>
        16
      </td>
    </tr>
    <tr>
      <td>
        21
      </td>
      <td>
        John L. Sandlin
      </td>
      <td>
        34242
      </td>
      <td>
        25
      </td>
    </tr>
    <tr>
      <td>
        33
      </td>
      <td>
        Dorothy J. Carroll
      </td>
      <td>
        42342
      </td>
      <td>
        18
      </td>
    </tr>
    <tr>
      <td>
        32
      </td>
      <td>
        Thomas M. Hawkins
      </td>
      <td>
        42342
      </td>
      <td>
        5
      </td>
    </tr>
    <tr>
      <td>
        42
      </td>
      <td>
        Susan D. Jimenez
      </td>
      <td>
        45345
      </td>
      <td>
        10
      </td>
    </tr>
    <tr>
      <td>
        43
      </td>
      <td>
        Margarita D. Lambert
      </td>
      <td>
        52342
      </td>
      <td>
        21
      </td>
    </tr>
    <tr>
      <td>
        30
      </td>
      <td>
        Stacey D. Ward
      </td>
      <td>
        64563
      </td>
      <td>
        24
      </td>
    </tr>
    <tr>
      <td>
        39
      </td>
      <td>
        Felix M. Seeber
      </td>
      <td>
        103596
      </td>
      <td>
        11
      </td>
    </tr>
    <tr>
      <td>
        17
      </td>
      <td>
        Camille B. Burton
      </td>
      <td>
        654645
      </td>
      <td>
        15
      </td>
    </tr>
    <tr>
      <td>
        35
      </td>
      <td>
        George K. Pancoast
      </td>
      <td>
        755463
      </td>
      <td>
        2
      </td>
    </tr>
    <tr>
      <td>
        18
      </td>
      <td>
        Tommye R. Robinson
      </td>
      <td>
        5345345
      </td>
      <td>
        4
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Rank is sorted correctly 1`] = `
<table
  class="table table-striped"
  data-testid="leaderboard-table"
>
  <thead>
    <tr>
      <th>
        Age
      </th>
      <th>
        Name
      </th>
      <th>
        Points
      </th>
      <th>
        Rank
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        25
      </td>
      <td>
        Sharon R. Ryan
      </td>
      <td>
        134
      </td>
      <td>
        1
      </td>
    </tr>
    <tr>
      <td>
        35
      </td>
      <td>
        George K. Pancoast
      </td>
      <td>
        755463
      </td>
      <td>
        2
      </td>
    </tr>
    <tr>
      <td>
        41
      </td>
      <td>
        Marlana J. Huff
      </td>
      <td>
        3123
      </td>
      <td>
        3
      </td>
    </tr>
    <tr>
      <td>
        18
      </td>
      <td>
        Tommye R. Robinson
      </td>
      <td>
        5345345
      </td>
      <td>
        4
      </td>
    </tr>
    <tr>
      <td>
        32
      </td>
      <td>
        Thomas M. Hawkins
      </td>
      <td>
        42342
      </td>
      <td>
        5
      </td>
    </tr>
    <tr>
      <td>
        29
      </td>
      <td>
        Arlene R. Grimes
      </td>
      <td>
        7675
      </td>
      <td>
        6
      </td>
    </tr>
    <tr>
      <td>
        10
      </td>
      <td>
        Michael E. Rogers
      </td>
      <td>
        274
      </td>
      <td>
        7
      </td>
    </tr>
    <tr>
      <td>
        22
      </td>
      <td>
        Joseph J. Freeman
      </td>
      <td>
        354
      </td>
      <td>
        8
      </td>
    </tr>
    <tr>
      <td>
        15
      </td>
      <td>
        Elizabeth R. Olsen
      </td>
      <td>
        452
      </td>
      <td>
        9
      </td>
    </tr>
    <tr>
      <td>
        42
      </td>
      <td>
        Susan D. Jimenez
      </td>
      <td>
        45345
      </td>
      <td>
        10
      </td>
    </tr>
    <tr>
      <td>
        39
      </td>
      <td>
        Felix M. Seeber
      </td>
      <td>
        103596
      </td>
      <td>
        11
      </td>
    </tr>
    <tr>
      <td>
        37
      </td>
      <td>
        Alice J. Goldsmith
      </td>
      <td>
        105
      </td>
      <td>
        12
      </td>
    </tr>
    <tr>
      <td>
        13
      </td>
      <td>
        Joan T. Covert
      </td>
      <td>
        3124
      </td>
      <td>
        13
      </td>
    </tr>
    <tr>
      <td>
        11
      </td>
      <td>
        Richard M. Mass
      </td>
      <td>
        312
      </td>
      <td>
        14
      </td>
    </tr>
    <tr>
      <td>
        17
      </td>
      <td>
        Camille B. Burton
      </td>
      <td>
        654645
      </td>
      <td>
        15
      </td>
    </tr>
    <tr>
      <td>
        40
      </td>
      <td>
        Robert O. Lambert
      </td>
      <td>
        18978
      </td>
      <td>
        16
      </td>
    </tr>
    <tr>
      <td>
        19
      </td>
      <td>
        Grace C. Smith
      </td>
      <td>
        342
      </td>
      <td>
        17
      </td>
    </tr>
    <tr>
      <td>
        33
      </td>
      <td>
        Dorothy J. Carroll
      </td>
      <td>
        42342
      </td>
      <td>
        18
      </td>
    </tr>
    <tr>
      <td>
        24
      </td>
      <td>
        John R. Green
      </td>
      <td>
        6546
      </td>
      <td>
        19
      </td>
    </tr>
    <tr>
      <td>
        23
      </td>
      <td>
        Joseph S. Perez
      </td>
      <td>
        31
      </td>
      <td>
        20
      </td>
    </tr>
    <tr>
      <td>
        43
      </td>
      <td>
        Margarita D. Lambert
      </td>
      <td>
        52342
      </td>
      <td>
        21
      </td>
    </tr>
    <tr>
      <td>
        20
      </td>
      <td>
        Jennifer C. West
      </td>
      <td>
        213
      </td>
      <td>
        22
      </td>
    </tr>
    <tr>
      <td>
        65
      </td>
      <td>
        Louis E. McCrae
      </td>
      <td>
        1241
      </td>
      <td>
        23
      </td>
    </tr>
    <tr>
      <td>
        30
      </td>
      <td>
        Stacey D. Ward
      </td>
      <td>
        64563
      </td>
      <td>
        24
      </td>
    </tr>
    <tr>
      <td>
        21
      </td>
      <td>
        John L. Sandlin
      </td>
      <td>
        34242
      </td>
      <td>
        25
      </td>
    </tr>
  </tbody>
</table>
`;
